kcp/pkg/util/watcher_util.go

```diff
@@ -263,11 +276,11 @@ func PerformConfigMapCheck(ctx context.Context, reader client.Reader,
 }
 
 func PerformIstioGWCheck(ctx context.Context, istioClientSet *istioclient.Clientset,
```

Comment:
Could you remove this function and unify logic with HandleProcessing logic from the reconciler. They both do the same thing!

Reply:
I removed this function as the gateway resource will not change.